cocos/gfx/device-manager.ts

```diff
@@ -121,31 +136,24 @@ export class DeviceManager {
                 if (sys.browserType === BrowserType.UC) {
                     useWebGL2 = false;
                 }
-
-                const deviceCtors: Constructor<Device>[] = [];
-                if (WEBGPU) {
-                    deviceCtors.push(cclegacy.WebGPUDevice);
+                Device.canvas = canvas!;
+                if (WEBGPU || this._renderType === RenderType.WEBGPU) {
```

Comment:
Why should check WEBGPU and this._renderType? Does it mean there are two ways to use webgpu?

Reply:
Yes, WEBGPU is a flag for the wasm version.

Reply:
Does the browser define it?

Reply:
No, it is our internal flag to indicate when WebGPU is running in a WASM environment.

Reply:
I think we can remove wasm support after the web version is completed.